xsos: init at 0.7.19

nixinator opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

nixinator committed:
xsos:init at 0.7.19

nixinator commented:
Xsos, nice little shell script, however i can't get the shell auto completion to work as mentioned in  
https://discourse.nixos.org/t/packaging-a-shell-script-that-has-completions/10290/3

so a little help appreciated.

legendofmiracles commented on pkgs/os-specific/linux/xsos/default.nix:
```diff
@@ -0,0 +1,54 @@
+  
+stdenv.mkDerivation rec {
+  pname = "xsos";
+  version = "v0.7.19";
```
```suggestion
  version = "0.7.19";
```

legendofmiracles commented on pkgs/os-specific/linux/xsos/default.nix:
```diff
@@ -0,0 +1,54 @@
+  src = fetchFromGitHub {
+    owner = "ryran";
+    repo = "xsos";
+    rev = version;
```
```suggestion
    rev = "v${version}";
```

legendofmiracles commented on pkgs/os-specific/linux/xsos/default.nix:
```diff
@@ -0,0 +1,54 @@
+    maintainers = [ maintainers.nixinator ];
+  };
+}
+
```
```suggestion
```

legendofmiracles commented:
The bash completion isn't installed because you never call the postInstall phase from installPhase. I've fixed it for you in [this patch](http://0x0.st/-hoi.txt). I also added busybox as a dependency because otherwise xsos couldn't find `runlevel`

nixinator commented:
> The bash completion isn't installed because you never call the postInstall phase from installPhase. I've fixed it for you in [this patch](http://0x0.st/-hoi.txt). I also added busybox as a dependency because otherwise xsos couldn't find `runlevel`

thanks for that. Hopefully busybox doesn't pollute the path with different tools. I've seen this happen with installing stuff via nix-env , you think your getting a tool from core utils etc etc, but it's actually coming from busy box itself due to name collisions on the path... ordering is important here, and can be darn confusing..... Is there a version of runlevel that isn't the cut down, limited functionality that busy-box provides.

legendofmiracles commented:
The only other packages that provide `runlevel` are `sysvinit` are `molly-guard`

legendofmiracles commented:
Result of `nixpkgs-review pr 147771` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xsos</li>
  </ul>
</details>

nixinator commented:
well `nixshell -p nixpkgs-review` is rather nice isn't it. This force is strong with mic92

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/652
